### PR TITLE
[FLINK-17275] port core training exercises, descriptions, solutions and tests

### PR DESCRIPTION
**Pull request by NicoK**
This PR ports the core exercises from https://github.com/ververica/flink-training-exercises to this new flink-training repo.

- use gradle build tooling
- have Java and Scala versions (Scala is optional)
- add descriptions in appropriate markdown files (rendered with links to
  navigate on github)

-----

These exercises are:
1. Filtering a Stream (Ride Cleansing)
1. Stateful Enrichment (Rides and Fares)
1. Windowed Analytics (Hourly Tips)
1. `ProcessFunction` and Timers (Long Ride Alerts)

Note that this set is not the same as the originally available ones but the most up-to-date versions for the tutorials developed as part of https://issues.apache.org/jira/browse/FLINK-12639.

-----

Next PR: #2 

---
This change is [Reviewable](https://reviewable.io/reviews/apache/flink-training/1)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,257 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
 # Flink Training Exercises
```
Comment (tzulitai) [inline README.md]:
```suggestion
# Apache Flink Training Exercises
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,257 @@
...
+Once that’s done you should be able to open [`RideCleansingTest`](ride-cleansing/src/test/java/org/apache/flink/training/exercises/ridecleansing/RideCleansingTest.java) and successfully run this test.
+
+> **:information_source: Note for Scala users:** You will need to use IntelliJ with the JetBrains Scala plugin, and you will need to add a Scala 2.12 SDK to the Global Libraries section of the Project Structure. IntelliJ will ask you for the latter when you open a Scala file.
+
+### Download the data sets
+
+You will also need to download the taxi data files used in this training by running the following commands
+
+```bash
+wget http://training.ververica.com/trainingData/nycTaxiRides.gz
```
Comment (tzulitai) [inline README.md:112]: Do we plan to move these files to somewhere else (on the Flink website, maybe?) other than the original hosted location under Ververica?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,257 @@
...
+
+### Exercises, Tests, and Solutions
+
+Many of the exercises include an `...Exercise` class with most of the necessary boilerplate code for getting started, as well as a JUnit Test class (`...Test`) with a few tests for your implementation, and a `...Solution` class with a complete solution.
+
+> **:information_source: Note:** As long as your `...Exercise` class is throwing a `MissingSolutionException`, the provided JUnit test classes will ignore that failure and verify the correctness of the solution implementation instead.
+
+-----
+
+Now you are ready to begin with the first exercise in our [**Labs**](LABS-OVERVIEW.md).
```
Comment (tzulitai) [inline README.md]: I suggest to add a section about the license at the bottom of the README:

```
## License

The code in this repository is licensed under the Apache Software License 2.
```

File: hourly-tips/src/main/scala/org/apache/flink/training/exercises/hourlytips/scala/HourlyTipsExercise.scala [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.flink.api.java.utils.ParameterTool
+import org.apache.flink.streaming.api.TimeCharacteristic
+import org.apache.flink.streaming.api.scala._
+import org.apache.flink.training.exercises.common.sources.TaxiFareSource
+import org.apache.flink.training.exercises.common.utils.ExerciseBase._
+import org.apache.flink.training.exercises.common.utils.{ExerciseBase, MissingSolutionException}
+
+/**
+  * The "Hourly Tips" exercise of the Flink training
+  * (http://training.ververica.com).
```
Comment (tzulitai) [inline hourly-tips/src/main/scala/org/apache/flink/training/exercises/hourlytips/scala/HourlyTipsExercise.scala]: For my understanding: will this links eventually be updated to point to Flink docs?

**Review by tzulitai — APPROVED** [review]
Looks good to me, thank you for porting this so quickly @NicoK. +1 from my side.

I went through the main README and did a quick legal checks on the source license headers - they are all licensed to ASF, so that's ✅ .

I also:
- Built the project locally with Gradle ✅ 
- Imported the project into Intellij IDEA ✅ 

Only have some comments about some downloadable resources / links.

File: README.md (hunk 2 above)
Comment (NicoK) [reply README.md:112]: Actually, I was planning to rewrite them into a data generator to not have to worry about licensing and removing this additional download step. I just created https://issues.apache.org/jira/browse/FLINK-17293 for that.

File: hourly-tips/src/main/scala/org/apache/flink/training/exercises/hourlytips/scala/HourlyTipsExercise.scala (hunk 4 above)
Comment (NicoK) [reply hourly-tips/src/main/scala/org/apache/flink/training/exercises/hourlytips/scala/HourlyTipsExercise.scala]: thanks, I forgot changing that note (turns out, I only changed it in the Java files and ported Scala a bit later - I'll change it for them as well.

**Comment by alpinegizmo** [comment]
+1 

Looks good to me. Thanks, @NicoK, for taking this on!

**Pull request merged**